vamb/hier.py

```diff
@@ -0,0 +1,480 @@
+import collections
+import csv
+import itertools
+from typing import Callable, Collection, Dict, Hashable, List, Optional, Sequence, TextIO, Tuple
+
+import networkx as nx
+import numpy as np
+from numpy.typing import ArrayLike
+
+
+class Hierarchy:
+    """Hierarchy of nodes 0, ..., n-1."""
+
+    def __init__(self, parents):
+        n = len(parents)
+        assert np.all(parents[1:] < np.arange(1, n))
+        self._parents = parents
+
+    def num_nodes(self) -> int:
+        return len(self._parents)
+
+    def edges(self) -> List[Tuple[int, int]]:
```

Comment:
```suggestion
    def edges(self) -> list[tuple[int, int]]:
```
You can just use the type directly in recent versions of Python